### PR TITLE
Database abstraction layer (mysql implementation)

### PR DESCRIPTION
**Pull request by hongaar**
Includes #456 
Based on #381

### CODE REVIEW

**Comment by rudehn** [comment]
The mysql implementation will return a dict cursor instead of a tuple cursor. Might want to standardize that

**Comment by hongaar** [comment]
@rudehn 👍 see  https://github.com/chaosbot/Chaos/pull/456/commits/a1038b94686b04199f1b62eed193e44c292350be

File: settings.py [hunk 1]
```diff
@@ -95,3 +95,16 @@
...
+DB_CONFIG = {
+    "filename": "db.sqlite"
+}
+# MySQL example
+# DB_CONFIG = {
+#     "host ": "localhost",
+#     "user ": "chaos",
+#     "password ": "chaos",
+#     "db ": "db"
+# }
```
Comment (PlasmaPower) [inline settings.py:110]: Shouldn't we be using MySQL? Transitioning will be difficult.

**Comment by hongaar** [comment]
@PlasmaPower I'd prefer sqlite for now, because it's much faster and more portable for simply storing some simple state records.

@md678685 With MySQL you'll also have to wait for locks; the operations are just running in another process.

And with MySQL we'll get another server to monitor and another dependency to manage (plus a whole bunch of memory goes to waste).

**Review by syvb — APPROVED** [review]

**Comment by PlasmaPower** [comment]
@hongaar How would you recommend I approach creating the table? I could do it with the sqlite CLI and the migrations tool.

**Comment by rudehn** [comment]
@PlasmaPower Just a heads up, but I'm going to start working on integrating [peewee](https://github.com/coleifer/peewee) so we can abstract out the db type from our queries. Peewee supports both mysql and sqlite.

If you'd like to take a look into it, you're more than welcome. I think having an ORM would be useful.

**Comment by PlasmaPower** [comment]
@rudehn Okay, I might wait for that. Will you be introducing a migration strategy? It looks like that has functions for doing the migrations themselves, but not versioning or similar.

**Comment by rudehn** [comment]
I envisioned using the migrations tool and peewee's migration functions. I don't see any versioning either. Is that something we are wanting to do?

**Comment by PlasmaPower** [comment]
@rudehn I guess the existing tools will work, it's just that they're shell based. I forgot that we could run a separate Python script from them.

**Comment by rudehn** [comment]
You can go ahead and setup a python migration script to get started.

**Comment by PlasmaPower** [comment]
@rudehn there's a problem - you can't import `lib.db` from the a script in the migrations directory.

**Comment by rudehn** [comment]
maybe we need to add the root level to the path? Try

`
sys.path.insert(...)
`

**Comment by PlasmaPower** [comment]
That might work, but it's pretty hacky.

**Pull request merged**